ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java

```diff
@@ -52,96 +60,196 @@ private TopicStream() {
 
     private final KafkaAvroDeserializer avroDeserializer;
     private final String topicName;
-    private final DateFormat dateFormat =
-        SimpleDateFormat.getDateTimeInstance(3, 1, Locale.getDefault());
+    private final DateFormat dateFormat;
+
+    private Optional<Formatter> keyFormatter = Optional.empty();
+    private Optional<Formatter> valueFormatter = Optional.empty();
 
-    private Formatter formatter;
+    public RecordFormatter(
+        final SchemaRegistryClient schemaRegistryClient,
+        final String topicName
+    ) {
+      this(
+          schemaRegistryClient,
+          topicName,
+          SimpleDateFormat.getDateTimeInstance(3, 1, Locale.getDefault())
```

Comment:
nit (not your code): constants for `3` and `1` (no idea what these mean 😂 )

Reply:
neither do I! :D